torch/utils/data/_dataset.py

```diff
@@ -133,3 +137,34 @@ def __iter__(self):
 
     def __len__(self):
         return self.length
+
+
+# Dataset for decoding data in different format (eg, json, png, etc)
+# Yield tuple(pathname, decoded object)
+# args:
+#    input_dataset - iterable with tuple(pathname, binary stream or bytes)
+#    decoders      - user passed in decoders/handlers, if None, a set of default decoders will be initialized
+#    length        - a nominal length of the dataset
+class RoutedDecoderIterDataset(IterDataset):
```

Comment:
We sure that we want to expose `IterableDataset` and `XIterDataset`? Sounds extremely error-prune to me.

Reply:
@SsnL This is a prototype which is used for further design discussion on Dataset. Currently, the high level design is to expose different kinds of Dataset, yes, XIterDataset, and maybe further XMapDataset. Everything is dataset, sampler, shuffler, sharding, etc.  This is could be more error-prune, need careful design and implementation, but we do have lots of benefit on decoupling the current dataset into smaller pieces. We are still working on a design doc, you guys can comment once it is ready. cc @VitalyFedyunin

Reply:
I was just mostly talking about the inconsistent naming.

Reply:
Naming is under discussion as well.